cmd/sftp-server.go

```diff
@@ -120,6 +123,224 @@ var supportedMACs = []string{
 	"hmac-sha2-256-etm@openssh.com", "hmac-sha2-512-etm@openssh.com", "hmac-sha2-256", "hmac-sha2-512", "hmac-sha1", "hmac-sha1-96",
 }
 
+func sshPubKeyAuth(c ssh.ConnMetadata, key ssh.PublicKey) (*ssh.Permissions, error) {
+	return authenticateSSHConnection(c, key, nil)
+}
+
+func sshPasswordAuth(c ssh.ConnMetadata, pass []byte) (*ssh.Permissions, error) {
+	return authenticateSSHConnection(c, nil, pass)
+}
+
+func authenticateSSHConnection(c ssh.ConnMetadata, key ssh.PublicKey, pass []byte) (*ssh.Permissions, error) {
+	user, found := strings.CutSuffix(c.User(), "=ldap")
+	if found {
+		return processLDAPAuthentication(key, pass, user)
+	}
+
+	user, found = strings.CutSuffix(c.User(), "=svc")
+	if found {
+		goto internalAuth
+	}
+
+	user = c.User()
+
+	if globalIAMSys.LDAPConfig.Enabled() {
+		perms, _ := processLDAPAuthentication(key, pass, user)
+		if perms != nil {
+			return perms, nil
+		}
+	}
```

Comment:
`user = c.User()` shouldn't be needed.
Removes the goto.

```suggestion
	user, found = strings.CutSuffix(c.User(), "=svc")
	if !found {
		if globalIAMSys.LDAPConfig.Enabled() {
			perms, _ := processLDAPAuthentication(key, pass, user)
			if perms != nil {
				return perms, nil
			}
		}
	}
```

Reply:
I'm going to keep it as is.. I feel like it's more confusing putting the ldap block under the =svc check. Since ldap is a part of the normal auth flow and the =svc suffix is like a special case.